lib/solveEquation/stepThrough.js

```diff
@@ -211,19 +289,26 @@ function addSimplificationSteps(steps, equation, debug=false) {
   // update `equation` to have the new simplified left node
   if (steps.length > 0) {
     equation = EquationStatus.resetChangeGroups(
-      steps[steps.length - 1 ].newEquation);
+      steps[steps.length - 1].newEquation);
   }
 
   // the updated equation from simplifing the left side is the old equation
   // (ie the "before" of the before and after) for simplifying the right side.
   oldEquation = equation.clone();
 
   const rightSteps = simplifyExpressionNode(equation.rightNode, false);
+  const rightFactorSteps = factor(equation.rightNode, false);
   const rightSubSteps = [];
   for (let i = 0; i < rightSteps.length; i++) {
     const step = rightSteps[i];
     rightSubSteps.push(EquationStatus.addRightStep(equation, step));
   }
+
+  for (let i = 0; i < rightFactorSteps.length; i++) {
+    const step = rightFactorSteps[i];
+    rightSubSteps.push(EquationStatus.addRightStep(equation, step));
+  }
```

Comment:
I take it back - this is really weird

essentially you're collecting steps for `simplifyExpressionNode` on an equation, and then collecting steps for calling `factor` on the same original equation and then combining the steps together. You should only take steps from one of them - I believe I commented on this earlier too.